pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/kafka/KafkaStreamLevelConsumer.java

```diff
@@ -87,11 +88,12 @@ public GenericRow next(GenericRow destination) {
     return next(destination, null);
   }
 
+  // NOTE: currently metadata is not updated as the simple consumer does not expose the relevant information
   @Override
-  public GenericRow next(GenericRow destination, StreamMessageMetadata metadata) {
+  public GenericRow next(GenericRow destination, RowMetadata metadata) {
```

Comment:
Please remove this API and keep the old one (without passing in RowMetadata). If we have to return metadata from this method in the future, we can put metadata into GenericRow.